src/pages/index.js

```diff
@@ -41,48 +51,110 @@ const enableValidation = (config) => {
 };
 enableValidation(config);
 
-const createCard = (cardObject) => {
-  const card = new Card(cardObject, "#card__template", () => {
-    imagePopup.open(cardObject);
+const handleImageClick = (data) => {
+  imagePopup.open(data);
+};
+
+const handleDelete = (card) => {
+  confirmationPopup.open();
+  confirmationPopup.setSubmitAction(() => {
+    api
+      .deleteCard(card)
+      .then(() => {})
+      .catch((err) => console.error(err));
+    confirmationPopup.close();
+    card.handleTrashBtn();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**